op-challenger/game/fault/trace/outputs/provider.go

```diff
@@ -0,0 +1,81 @@
+package outputs
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/types"
+	"github.com/ethereum-optimism/optimism/op-service/client"
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+var (
+	GetStepDataErr      = fmt.Errorf("GetStepData not supported")
+	AbsolutePreStateErr = fmt.Errorf("AbsolutePreState not supported")
+	PreStateRequestErr  = fmt.Errorf("Requested trace index is before prestate block")
+)
+
+type OutputRollupClient interface {
+	OutputAtBlock(ctx context.Context, blockNum uint64) (*eth.OutputResponse, error)
+}
+
+// OutputTraceProvider is a [types.TraceProvider] implementation that uses
+// output roots for given L2 Blocks as a trace.
+type OutputTraceProvider struct {
```

Comment:
One nice to have is the assertion that `OutputTraceProvider` satisfies the `TraceProvider` interface. You can do this by adding the following line: `var _ TraceProvider = (*OutputTraceProvider)(nil)` which will cause a compilation failure if `OutputTraceProvider` doesn't implement `TraveProvider`.